neo-codap/src/case-table/case-table.tsx

```diff
@@ -123,6 +128,21 @@ export class CaseTable extends React.Component<ICaseTableProps, ICaseTableState>
     });
   }
 
+  addLocalCaseToTable() {
+    const {dataSet} = this.props;
+    if (!dataSet) {
+      return;
+    }
+
+    const newCase: ICase = {};
+    dataSet.attributes.forEach((attr) => {
+      newCase[attr.name] = this.localCase[attr.id];
+    });
+
+    this.localCase = {};
+    addCasesToDataSet(dataSet, [newCase]);
```

Comment:
You could use `addCanonicalCasesToDataSet()` here, in which case you wouldn't need the conversion loop above. Canonical cases use attribute IDs for property names, which is the format you're already using for the `localCase`. While you're at it, the only other use of `addCasesToDataSet()` in this file could/should probably be converted to `addCanonicalCases()` as well. (It's adding an empty object, so the format of the property names is irrelevant.)